### PR TITLE
Add AKS setup instruction

### PR DESCRIPTION
**Pull request by josephwnv**
documentation about how to create AKS for NIM deployment.

### CODE REVIEW

File: cloud-service-providers/azure/aks/prerequisites/images/UsageQuota.jpg
Comment (supertetelman) [inline cloud-service-providers/azure/aks/prerequisites/images/UsageQuota.jpg:1]: Please censor out these images a little better. Many internal components are highly visible here, nothing critical, but shouldn't include in official documentation.

File: cloud-service-providers/azure/aks/setup/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+az aks create -g <resource group name> -n <aks name> --location <location has desired GPU> --zones <zone has desired GPU> --generate-ssh-keys
+```
+
+## Create GPU nodepool
+
+```
+az aks nodepool add --resource-group <resource group name> --cluster-name <aks name> --name <nodepool name> --node-count 1 --skip-gpu-driver-install --node-vm-size <Desire VM type> --node-osdisk-size 2048 --max-pods 110
+```
+
+|    Desire VM type                                  | Llama-3-8b    | Llama-3-70b   |
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/setup/README.md:24]: Is this documented somewhere on the official NIM docs? Feels like if we publish this here it will quickly become out of date.

I would rather see us point towards the official support matrix and have this information added there if possible.

File: cloud-service-providers/azure/aks/setup/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+| Standard_ND96isr_H100_v5                           | Optimized     | Optimized     |
+
+
+## Connect to AKS
+
+```
+az aks get-credentials --resource-group <resource group name> --name <aks name>
+```
+
+## Install GPU Operator
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/setup/README.md:42]: Are there official GPU Operator installation docs we can point to in addition to the simple steps below?

File: cloud-service-providers/azure/aks/setup/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Setup AKS
+
+The keys to create AKS for NIM is to create proper GPU nodepool.  Details are in the following table.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/setup/README.md:3]: Can we re-word this to make it more clear how the user can use the below table when deploying NIM?

File: cloud-service-providers/azure/aks/prerequisites/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Pre-requirement
+
+The keys to create AKS for NIM is to create proper GPU nodepool.  The proper GPU nodepool should have up to date driver.  This is only achievable via a preview cli extension.
+We cannot create AKS via Azure Portal GUI and we need to create it using CLI.  We need to setup our terminal with a several cli to perform the AKS creation.  Following is the detail instructions.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/prerequisites/README.md:4]: Avoid using pronouns like we/you in the docs if possible. I need to create a proper style guide doc for the repo, but overall we have been trying to use the impersonal tense.

File: cloud-service-providers/azure/aks/prerequisites/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Login into Azure Portal and click the subscription you want to use.
+
+![](./images/AzurePortal.jpg)
+
+Click "Use + Quota" 
+
+![](./images/UsageQuota.jpg)
+
+Search for 'NDAMSv4_A100' or 'NDSH100v5' for llama3-70b or 'NCADS_A100_v4' 'NDASv4_A100' 'NCadsH100v5' for llama3-8b and looking for region has enough quota.
+If no region has enough quota make click the edit icon at the right to reqest it.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/prerequisites/README.md:17]: "make click the edit" -> "click the edit"

File: cloud-service-providers/azure/aks/prerequisites/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+```
+az extension add --name aks-preview
+az extension update --name aks-preview
+```
+
+For more detail, Please reference this [link.](https://learn.microsoft.com/en-us/azure/aks/draft)
+
+
+## Install NGC CLI
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/prerequisites/README.md:41]: Do we need to use the NGC CLI anywhere? Can we add a line here explaining what this would be for? Possibly point to the official CLI installation docs as well.

File: cloud-service-providers/azure/aks/prerequisites/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Pre-requirement
+
+The keys to create AKS for NIM is to create proper GPU nodepool.  The proper GPU nodepool should have up to date driver.  This is only achievable via a preview cli extension.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/prerequisites/README.md:3]: Can we reword this to say that the driver must meet the NIM minimal requirements and to say why the nodepool matters?

File: cloud-service-providers/azure/aks/README.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# nim on aks
+
+AKS need to be properly created in order to be able deploy NIM on it.  There are two things need to be taken care of.  Make sure you have right GPU for NIM and right driver version as well.  Unfortunately, default GPU driver in AKS usually always a little bit too old for latest NVidia software in general.  Microsoft has no official solution for it yet.  There is a preview version of CLI to solve this issue.  We need to create AKS using preview version of CLI.  Prerequisite section talks about how to set up your local environment to enable you to create AKS from preview version of CLI.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/README.md:3]: Can we spell out what AKS means at first mention? Is that usually how Azure docs work?

Comment (supertetelman) [inline cloud-service-providers/azure/aks/README.md:3]: Can we break this into two paragraphs and word it more passively. Something lik:

To deploy NIM on AKS successfully, ensure you have the right GPU and driver version. The default GPU driver in AKS is usually outdated for the latest NVIDIA software, and Microsoft does not yet have an official solution for this issue.

To resolve this, use the preview version of the CLI to create the AKS cluster. The Prerequisites section explains how to set up your local environment to enable AKS creation with the preview CLI.

File: cloud-service-providers/azure/aks/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+After you are ready to create AKS, Next thing is to choose the right GPU instance.  Only L40S, A100, H100 GPU work for NIM but not all flavor.  Create AKS section has more details about this.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/README.md:5]: Next -> "the next"

Comment (supertetelman) [inline cloud-service-providers/azure/aks/README.md:5]: flavor -> "system configurations"

**Review by supertetelman — CHANGES_REQUESTED** [review]
Several comments left around images and flow.

It would also be good if we could explain why we need to install each of the components we are adding.

Great first draft.

File: cloud-service-providers/azure/aks/prerequisites/images/UsageQuota.jpg
Comment (josephwnv) [reply cloud-service-providers/azure/aks/prerequisites/images/UsageQuota.jpg:1]: done

File: cloud-service-providers/azure/aks/setup/README.md (hunk 1 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/setup/README.md]: done

File: cloud-service-providers/azure/aks/prerequisites/README.md (hunk 5 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/prerequisites/README.md]: Done

File: cloud-service-providers/azure/aks/prerequisites/README.md (hunk 6 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/prerequisites/README.md:23]: I will remove it.  I need that when I try to download dataset for customizer to create lora.  That is our of scope.  maybe later I when I make for nemo on AKS.

File: cloud-service-providers/azure/aks/README.md (hunk 9 above)
Comment (supertetelman) [reply cloud-service-providers/azure/aks/README.md:5]: Could we fix these two typos before merge?

File: cloud-service-providers/azure/aks/setup/README.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
+# Setup Azure Kubernetes Service (AKS)
+
+The keys to create Azure Kubernetes Service (AKS) for NIM is to create proper GPU nodepool.  The following steps guide you how to find it.
```
Comment (supertetelman) [inline cloud-service-providers/azure/aks/setup/README.md:3]: The keys -> "the key to creating ...."

**Comment by supertetelman** [comment]
There are still a few typos/confusing verbiages I pointed out. Let's fix those real quick before we merge these docs through.

File: cloud-service-providers/azure/aks/setup/README.md (hunk 10 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/setup/README.md:3]: Not applicable now

File: cloud-service-providers/azure/aks/README.md (hunk 9 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/README.md]: done

Comment (josephwnv) [reply cloud-service-providers/azure/aks/README.md]: done

File: cloud-service-providers/azure/aks/README.md (hunk 8 above)
Comment (josephwnv) [reply cloud-service-providers/azure/aks/README.md]: done

Comment (josephwnv) [reply cloud-service-providers/azure/aks/README.md]: done

**Review by supertetelman — APPROVED** [review]
